### ISSUE TITLE
Compile times on website are way too long

### ISSUE THREAD

**Issue by b-studios** [effekt-lang/effekt#62]
This can be observed by visiting

  https://effekt-lang.org/docs/casestudies/smc

then in the run dialog on the bottom of the page enter `()` and click run again. I observe compile times of around 10second. Also clicking a second time doesn't speed up, which indicates that caching is not working...

**Comment by b-studios** [comment]
I could trace the problem to the way files are (re-) compiled in the webbrowser. 

Files are compiled "dynamically", when imported.

On a page like the smc case study, we have many small editor windows. Every window imports the standard library and potentially other libraries. Due to some oversights in the implementation it appears the files are actually recompiled every time!
```
compiling effekt
compiling immutable/list
compiling effekt
compiling immutable/option
compiling effekt
compiling immutable/option
compiling effekt
compiling mutable/array
compiling effekt
compiling immutable/option
compiling effekt
compiling unsafe/cont
compiling effekt
compiling module0
compiling effekt
compiling immutable/list
compiling effekt
compiling immutable/option
compiling effekt
...
```

**Comment by b-studios** [comment]
This is closed by improved caching in #79
